src/tests/EventPipeStress/Orchestrator/Program.cs

```diff
@@ -0,0 +1,297 @@
+            var testResults = new TestResults(eventSize);
+
+            Func<int, TestResult> threadProc = readerType switch
+            {
+                ReaderType.Stream => UseFS(rundown, bufferSize),
+                ReaderType.EventPipeEventSource => UseEPES(rundown, bufferSize, slowReader),
+                _ => throw new ArgumentException("Invalid reader type")
+            };
+
+            if (eventRate == -1 && burstPattern != BurstPattern.NONE)
+                throw new ArgumentException("Must have burst pattern of NONE if rate is -1");
+
+            Console.WriteLine($"Configuration: event_size={eventSize}, event_rate={eventRate}, cores={cores}, num_threads={threads}, reader={readerType}, event_rate={(eventRate == -1 ? -1 : eventRate * threads)}, burst_pattern={burstPattern.ToString()}, slow_reader={slowReader}, duration={duration}");
+
+            for (int iteration = 0; iteration < iterations; iteration++)
+            {
+                Console.WriteLine("========================================================");
+                Console.WriteLine($"Starting iteration {iteration + 1}");
+
+                Process eventWritingProc = new Process();
+                eventWritingProc.StartInfo.FileName = stressPath.FullName;
+                eventWritingProc.StartInfo.Arguments = $"--threads {(threads == -1 ? cores.ToString() : threads.ToString())} --event-count {eventCount} --event-size {eventSize} --event-rate {eventRate} --burst-pattern {burstPattern} --duration {(int)durationTimeSpan.TotalSeconds}";
+                eventWritingProc.StartInfo.UseShellExecute = false;
+                eventWritingProc.StartInfo.RedirectStandardInput = true;
+                eventWritingProc.StartInfo.Environment["COMPlus_StressLog"] = "1";
+                eventWritingProc.StartInfo.Environment["COMPlus_LogFacility"] = "2000";
+                eventWritingProc.StartInfo.Environment["COMPlus_LogLevel"] = "8";
+                eventWritingProc.StartInfo.Environment["COMPlus_StressLogSize"] = "0x1000000";
+                eventWritingProc.Start();
+
+                Console.WriteLine($"Executing: {eventWritingProc.StartInfo.FileName} {eventWritingProc.StartInfo.Arguments}");
+                if (!RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+                {
+                    // Set affinity and priority
+                    ulong affinityMask = 0;
+                    for (int j = 0; j < cores; j++)
+                    {
+                        affinityMask |= ((ulong)1 << j);
+                    }
+                    eventWritingProc.ProcessorAffinity = (IntPtr)((ulong)eventWritingProc.ProcessorAffinity & affinityMask);
+                    eventWritingProc.PriorityClass = ProcessPriorityClass.RealTime; // Set the process priority to highest possible
+                }
+
+                // Start listening to the event.
+                Task<TestResult> listenerTask = Task.Run(() => threadProc(eventWritingProc.Id), ct);
```

Comment:
Can you just use a `Thread` here instead of a Task? It's usually better to use a dedicated thread with highest priority possible set to keep the measurements consistent when benchmarking things - I understand this is the listener side but we've both seen listeners being the bottleneck in these benchmarks :p

Reply:
Since the main thread should be waiting the entire time that task is running, I think it's probably fine to run it in a task.  If we want to use a thread, though, I guess we could use a `TaskCompletionSource` from the inside the thread to give us `await` semantics still.